File: CPP/Algorithm/Substring_or_Subarray.cpp
// Complexity: O(n^2)
// Substring:

auto sstr(auto &s) {
    int64_t n = s.size(), k = 0;
    vector<string> sub((n * (n + 1)) >> 1);
    for (int i = 0; i < n; ++i) {
        string tmp;
        for (int j = i; j < n; ++j) {
            tmp += s[j]; sub[k++] = tmp;
        }
    }
    return sub;
}

// Subarray:

auto sar(auto &v) {
    int64_t n = v.size(), k = 0;
    vector<vector<int64_t>> sub((n * (n + 1)) >> 1);
    for (int i = 0; i < n; ++i) {
        vector<int64_t> tmp;
        for (int j = i; j < n; ++j) {
            tmp.emplace_back(v[j]); sub[k++] = tmp;
        }
    }
    return sub;
}

File: Python/Algorithm/Substring_or_Subarray.py
# Complexity: O(n^2)
# Substring:

def sstr(s):
    n = len(s); sub = []
    for i in range(n):
        tmp = ""
        for j in range(i, n):
            tmp += s[j]; sub.append(tmp)
    return sub

# Subarray:

def sar(v):
    n = len(v); sub = []
    for i in range(n):
        tmp = []
        for j in range(i, n):
            tmp.append(v[j]); sub.append(tmp.copy())
    return sub